### DISCUSSION TITLE
How to go from a GeoJSON Feature to a set of geohashes?

### DISCUSSION THREAD

**Discussion by lorenzofelletti** [georust/geo#1009] [Q&A]
First of all, I hope that I am posting in the right place (I apologize if I am not).

I have a GeoJSON file that contains a polygon for each districts of a city. What I would like to get is, for each district, the set of geohashes that are contained within the district's polygon. To decode the GeoJSON I'm using the `geojson` crate, for the geohashes the `geohash` create, but I did not found in the documentation a way to do something like: 
- from a `geojson::geometry::Geometry`
```Rust
geometry.geohashes_inside(precision)
```
Does someone has already faced something similar, and can point me out some resources to be able to find a solution for my problem?

**Comment by lnicola** [comment]
I don't think this belongs to `geo`, but neither does it belong to `geohash` as far as I can see.

I believe the best option is to define a grid of the resolution you want, pick the squares that intersect your geometry, then encode their centres using `geohash`.

**Comment by lorenzofelletti** [comment]
Sorry for mis-posting and the late response.
First, thank you for your suggestion!
Second, in case this could be helpful to someone in the future, I found something implementing just the functionality I needed [here](https://github.com/worace/geoq/blob/master/src/geoq/geohash.rs).
